### PR TITLE
Added flag to skip foreign key creation for the CLI

### LINKED ISSUE #145: Improve performance of Foreign Key processing
The current behavior in HarbourBridge is to process schema change statements for creation of foreign keys in a linear order. Each schema update for foreign key creation can take a long time in Cloud Spanner since it needs to do data validation as well as create backing indexes for the foreign key, where applicable. For a source database with a large number of foreign keys, this can become problematic since each foreign key creation takes a long time and the overall time taken can be in hours. This can be solved using the following approaches:

### Alternative 1.1: Batch creation for multiple foreign key in the same schema update. 
There are some caveats to the same though:
#### Caveat 1.2.1: If there are too many schema updates happening at the same time, Cloud Spanner may throttle some of them. See https://cloud.google.com/spanner/docs/schema-updates#frequency for more details.

We will need to do some investigation and come up with a heuristic to combine Alternative 1.1 and Alternative 1.2 for optimal performance.

Foreign keys in HarbourBridge are today created after all the data has already migrated. This is done to avoid the issue where referencing rows are written before the referenced rows when migrating the data. This can be solved by doing the data migration in the same order as the one implied by DAG created by following foreign key dependencies. However, this is not an option when using mysqldump or pgdump since those cannot be read out of order to migrate tables in a specific order.

## Problem Statement 2: Create foreign keys during data migration for direct connect mode
There is some investigation/experimentation needed to understand if this would actually help with the overall data migration time.

For some of the customers, it may make sense to not have HarbourBridge perform the foreign key creations in the first place if they care more about reducing the down time and are willing to apply foreign keys themselves after the migration is over. To support such use cases, we should also provide a command line flag to skip foreign key creation during data migration. This is specially important if the data migration is followed by CDC migration since foreign keys may cause the migration of CDC change events to fail and get retried multiple times and thus increase the overall application downtime during migration.

### PR DESCRIPTION
**Pull request by Deep1998**
Throw an exception when both schema-only and skip-foreign-keys are set.
This flag does not omit the Foreign Key statements in the DDL.

Fixes 145 part 3

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -54,6 +55,7 @@ func init() {
 	flag.BoolVar(&verbose, "v", false, "verbose: print additional output")
 	flag.BoolVar(&schemaOnly, "schema-only", false, "schema-only: in this mode we do schema conversion, but skip data conversion")
 	flag.BoolVar(&dataOnly, "data-only", false, "data-only: in this mode we skip schema conversion and just do data conversion (use the session flag to specify the session file for schema and data mapping)")
+	flag.BoolVar(&skipForeignKeys, "skip-foreign-keys", false, "skip-foreign-keys: in this mode we do not update the foreign keys after data migration is complete")
```
Comment (agasheesh) [inline main.go:58]: this does not specify a mode unlike dataOnly and schemaOnly - how about we say something like 

"if true, skip creating foreign keys after data migration is complete (ddl statements for foreign keys can still be found in the downloaded schema.ddl.json file and the same can be applied separately)"

**Review by agasheesh — APPROVED** [review]
Looks good. Thanks for all the changes.

File: README.md [hunk 2]
```diff
@@ -284,6 +284,10 @@ Any data in the source database will be ignored.
 A spanner database will be created based on the schema state provided
 by a session file (`-session`) and data will be migrated.
 
+`-skip-foreign-keys` Specifies that foreign key creation will be skipped after 
+data migration is complete. It cannot be used with schema-only mode. This flag does not
```
Comment (nevinheintze) [inline README.md:288]: Suggest a slight rewording:

`-skip-foreign-keys` Controls whether we add foreign key constraints after
data migration is complete. This flag cannot be used with schema-only mode,
and does not affect the generation of foreign key statements during schema
processing i.e. foreign key constraints will still appear in the generated Spanner
DDL files.

File: testing/mysql/integration_test.go [hunk 3]
```diff
@@ -118,7 +118,7 @@ func TestIntegration_MYSQLDUMP_SimpleUse(t *testing.T) {
 	if err != nil {
```
Comment (nevinheintze) [inline testing/mysql/integration_test.go:118]: We should add a simple test for skip-foreign-keys -- maybe a follow-up PR?

**Review by nevinheintze — APPROVED** [review]
Really sorry for the slow review.
Just a couple of small comments.

Looks good -- thanks!

File: README.md (hunk 2 above)
Comment (Deep1998) [reply README.md]: Updated

File: testing/mysql/integration_test.go (hunk 3 above)
Comment (Deep1998) [reply testing/mysql/integration_test.go:118]: Agreed, I'll put it up in the next PR.

**Pull request merged**